src/app/clusters/electrical-power-measurement-server/electrical-power-measurement-server.h

```diff
@@ -0,0 +1,113 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/core/Optional.h>
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ElectricalPowerMeasurement {
+
+using namespace chip::app::Clusters::ElectricalPowerMeasurement::Attributes;
+using namespace chip::app::Clusters::ElectricalPowerMeasurement::Structs;
+
+using chip::Protocols::InteractionModel::Status;
+
+class Delegate
+{
+public:
+    virtual ~Delegate() = default;
+
+    void SetEndpointId(EndpointId aEndpoint) { mEndpointId = aEndpoint; }
+
+    virtual PowerModeEnum GetPowerMode()                                                                         = 0;
+    virtual uint8_t GetNumberOfMeasurementTypes()                                                                = 0;
+    virtual DataModel::List<const Structs::MeasurementAccuracyStruct::Type> GetAccuracy()                        = 0;
```

Comment:
How is the delegate supposed to implement this?  Once it returns this List it can never free the backing buffer, right?  This seems like a pretty annoying "keep it all in RAM forever" requirement, especially for non-Fixed things like Ranges or HarmonicCurrents/HarmonicPhases (for this last, seems like it's basically a memory leak).